lib/bn.js

```diff
@@ -56,7 +56,7 @@
 
   BN.isBN = function isBN (num) {
     return num !== null && typeof num === 'object' &&
-      num.constructor.name === 'BN' && Array.isArray(num.words);
+      num.constructor.name === BN.name && Array.isArray(num.words);
```

Comment:
Well that seems to have good browser support :)

The question is still there: that will still reject BN instances of different versions.  I am pretty sure that in some browserified instances that will happen.  I think it is the desired behaviour, but I don't think many libraries are prepared for this.